src/layer/AtmosphereLayer.js

```diff
@@ -0,0 +1,414 @@
+    ],
+    function (ArgumentError,
+              GroundProgram,
+              Layer,
+              Logger,
+              Matrix,
+              Matrix3,
+              Sector,
+              SkyProgram,
+              Vec3) {
+        "use strict";
+
+        /**
+         * Constructs a layer showing the Earth's atmosphere.
+         * @alias AtmosphereLayer
+         * @constructor
+         * @classdesc Provides a layer showing the Earth's atmosphere.
+         * @param {URL} nightImageSource optional url for the night texture.
+         * @augments Layer
+         */
+        var AtmosphereLayer = function (nightImageSource) {
+
+            Layer.call(this, "Atmosphere");
+
+            //image for the night texture
+            this._nightImageSource = nightImageSource || '../images/dnb_land_ocean_ice_2012.png';
+
+            // Documented in defineProperties below.
+            this._lightLocation = null;
+
+            //the light direction in cartesian space, computed form the lightLocation or defaults to the eyePoint
+            this.activeLightDirection = new Vec3(0, 0, 0);
+
+            this.pickEnabled = false;
+
+            this._fullSphereSector = Sector.FULL_SPHERE;
+
+            // Internal use only. Intentionally not documented.
+            this.skyData = {};
+
+            // Documented in defineProperties below.
+            this._skyWidth = 128;
+
+            // Documented in defineProperties below.
+            this._skyHeight = 128;
+
+            // Documented in defineProperties below.
+            this._skyPoints = null;
+
+            // Documented in defineProperties below.
+            this._skyTriStrip = null;
+
+            //texture coordinate matrix used for the night texture
+            this._texMatrix = Matrix3.fromIdentity();
+
+            //modelViewProjection matrix
+            this._mvpMatrix = Matrix.fromIdentity();
+
+            //the night texture
+            this._activeTexture = null;
+        };
+
+        AtmosphereLayer.prototype = Object.create(Layer.prototype);
+
+        Object.defineProperties(AtmosphereLayer.prototype, {
+
+            /**
+             * The geographic location of the light (sun).
+             * @memberof AtmosphereLayer.prototype
+             * @type {Position}
+             */
+            lightLocation: {
+                get: function () {
+                    return this._lightLocation;
+                },
+                set: function (value) {
+                    this._lightLocation = value;
+                }
+            },
+
+            /**
+             * Url for the night texture.
+             * @memberof AtmosphereLayer.prototype
+             * @type {URL}
+             */
+            nightImageSource: {
+                get: function () {
+                    return this._nightImageSource;
+                },
+                set: function (value) {
+                    this._nightImageSource = value;
+                }
+            },
+
```

Comment:
The properties skyWidth, skyHeight, skyPoints and skyTriStrip are considered internal implementation details. Please mark them as internal in the property declarations above, and remove their associated property accessors.